src/org/daisy/dotify/formatter/impl/obfl/ObflParserImpl.java

```diff
@@ -2134,6 +2143,18 @@ private boolean getHyphenate(XMLEvent event, boolean hyphenate) {
         return hyphenate;
     }
 
+    private boolean getMarkCapitalLetters(XMLEvent event, boolean markCapitalLetters) {
+        String markCL = getAttr(event, ObflQName.ATTR_MARK_CAPITAL_LETTERS);
+        if (markCL != null) {
+            if ("".equals(markCL)) {
+                return markCapitalLettersGlobal;
+            } else {
+                return "true".equals(markCL);
+            }
+        }
+        return markCapitalLetters;
```

Comment:
What I miss is some explanation or JavaDoc of how this markCapitalLetters works.
Apparently there is a global or a config setting.
And, apparently you can set the value of markCapitalLetters to true, to false or to the global config setting. The latter is done by providing an empty attribute value.
It would also be very nice to have an explanation of the word "markCapitalLetters" somewhere in JavaDoc - it seems to me that if this property is true, then the output may contain capital letters, otherwise the output contains only lowercase letters.
I know that the OBFL must still be done, but some more explanation in the form of JavaDoc would be appreciated.

Reply:
Hi @PaulRambags 

Then you exactly understood how it works. If you have a user who doesn't know how to interpret marks for capitalization, we will make everything lowercase so the system will not add any marks.

Capitalization is done in Swedish braille by adding a braille char before the capitalized char, or two before a capitalized word or three before a string that is capitalized, and then you have three marks to "turn off" capitalization again.

Other translations could have other marks that indicate capitalization for the user, so how this function works depends on the Braille system.

"Turning off" the feature is simple though, make everything lowercase.

So are you looking for an explanation in the code that we will make the text lowercase if false, or do you want the full explanation that might be misleading in some cases?

Best regards
Daniel

Reply:
I was looking for an explanation in the code, as comments or as Javadoc.
What I miss is:

- An explanation of the word "markCapitalLetters": if markCapitalLetters is set to false, then the output won't contain any capitalization characters. In other words, the output will be in lower case.
- An explanation of the usage of the attribute `mark-capital-letters`. Apparently this attribute can contain `true`, `false` or the empty string. The empty string is a reset to the global value.
- An explanation of how to set this global value.